core/services/pipeline/http_shared.go

```diff
@@ -0,0 +1,86 @@
+package pipeline
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+func makeHTTPRequest(
+	ctx context.Context,
+	method StringParam,
+	url URLParam,
+	requestData MapParam,
+	allowUnrestrictedNetworkAccess BoolParam,
+	cfg Config,
+) ([]byte, time.Duration, error) {
+
+	var bodyReader io.Reader
+	if requestData != nil {
+		bodyBytes, err := json.Marshal(requestData)
+		if err != nil {
+			return nil, 0, errors.Wrap(err, "failed to encode request body as JSON")
+		}
+		bodyReader = bytes.NewReader(bodyBytes)
+	}
+
+	request, err := http.NewRequest(string(method), url.String(), bodyReader)
+	if err != nil {
+		return nil, 0, errors.Wrap(err, "failed to create http.Request")
+	}
+	request.Header.Set("Content-Type", "application/json")
+
+	config := utils.HTTPRequestConfig{
+		Timeout:                        cfg.DefaultHTTPTimeout().Duration(),
+		MaxAttempts:                    cfg.DefaultMaxHTTPAttempts(),
+		SizeLimit:                      cfg.DefaultHTTPLimit(),
+		AllowUnrestrictedNetworkAccess: bool(allowUnrestrictedNetworkAccess),
```

Comment:
Out of scope of this PR, but we need to remove this and replace with a custom URL type - like TrustedURL and UntrustedURL